### ISSUE TITLE
Figure out what methods should exist between the app and the worker!

### ISSUE THREAD

**Issue by eddieantonio** [eddieantonio/predictive-text-studio#108]
Right now `compileModel()` and `saveFile()` are wonky. Addtionally, there probably should be events that fire and update stores on the front-end.

Please ask me, @eddieantonio, to clarify my thoughts on this issue, but I think it should be addressed sooner rather than later! With tests!

**Comment by xqiu1** [comment]
@eddieantonio Eddie, for this issue, do you mean adding some functions to storage that may be needed later? Similar to what we did in packageInfo.

**Comment by eddieantonio** [comment]
> @eddieantonio Eddie, for this issue, do you mean adding some functions to storage that may be needed later? Similar to what we did in packageInfo.

No, I mean establishing **properly justified**, **well-documented** methods in [`predictive-text-worker.ts`](https://github.com/eddieantonio/predictive-text-studio/blob/e3ad0e33c8ca7878712778ec392be356ea8551c6/src/common/predictive-text-studio-worker.ts).

This means figuring out what tasks the app should expect be done within the worker. In particular, I think `saveFile()` and `compileModel()` are (in their current state), poorly-defined, and a bit strange. For example, `saveFile()` not only saves a file, but also compiles the model, and generates the KMP file. `compileModel()` returns some JavaScript code as a string, that is not actually used anywhere.

Here are some tasks that I think are useful:

 - `addDictionarySourceToProject(name: string, contents: File): Promise<number>` —  takes a dictionary source, e.g., from an uploaded Excel file, and stores it; it should return how many words were added
 - `onPackageCompileStart(callback: () => void): void` — register a callback that is called directly before the KMP package is generated. This is useful for showing a progress bar in the UI (e.g., _Compile in progress_)
 - `onPackageCompileError(callback: (err: Error) => void): void` — register a callbcak that is called if the KMP package compilation fails
 - `onPackageCompileSuccess(callback: (kmp: Blob) => void): void` — register a callback that is called when the KMP package has succesfully compiled.
 - `setProjectData(metadata: {}): void` sets optional and required metadata such as BCP-47, language name, author name, copyright string, etc.

The interface doesn't _strictly_ need to adhere to what I wrote, but these are my ideas.

The reason I don't want a `compileModel()` method, is because I want the package to automatically be recompiled, and be available to download before the user even thinks about hitting the download button!

**EDIT**: the existing methods `saveFile()` and `compileModel()` should be marked as deprecated, and deleted in the future
